src/items/galleon.rs

```diff
@@ -0,0 +1,318 @@
+        );
+
+        (pos, facing)
+    }
+
+    pub fn update(&mut self) -> bool {
+        self.pos += self.speed * get_frame_time();
+
+        {
+            let resources = storage::get::<Resources>();
+            let map_width = (resources.tiled_map.raw_tiled_map.tilewidth
+                * resources.tiled_map.raw_tiled_map.width) as f32;
+            if self.pos.x + Self::WIDTH < -map_width * 0.5 || self.pos.x > map_width * 1.5 {
+                return false;
+            }
+        }
+
+        for mut player in scene::find_nodes_by_type::<crate::nodes::Player>() {
+            if player.get_hitbox().overlaps(&Rect::new(
+                self.pos.x,
+                self.pos.y,
+                Self::WIDTH,
+                Self::HEIGHT,
+            )) && player.id != self.owner_id
+                && !player.dead
+            {
+                scene::find_node_by_type::<crate::nodes::Camera>()
+                    .unwrap()
+                    .shake_noise(1.0, 10, 1.);
+                {
+                    let mut resources = storage::get_mut::<Resources>();
+                    resources.hit_fxses.spawn(player.body.pos);
+                }
+                let direction = self.pos.x > (player.body.pos.x + 10.);
+                player.kill(direction);
+            }
+        }
+
+        true
+    }
+
+    pub fn draw(&self, pos: Vec2, facing: bool) {
+        draw_texture_ex(
+            self.sprite,
+            pos.x,
+            pos.y,
+            color::WHITE,
+            DrawTextureParams {
+                dest_size: Some(Vec2::new(Self::WIDTH, Self::HEIGHT)),
+                flip_x: facing,
+                ..Default::default()
+            },
+        );
+    }
+}
+
+impl Node for FlyingGalleon {
+    fn draw(node: RefMut<Self>) {
+        node.draw(node.pos, node.facing);
+    }
+
+    fn fixed_update(mut node: RefMut<Self>) {
+        if !node.update() {
+            node.delete();
+        }
+    }
+}
+
+pub struct Galleon {
+    sprite: GunlikeAnimation,
+
+    body: PhysicsBody,
+    throwable: ThrowableItem,
+}
+
+impl Galleon {
+    pub const COLLIDER_WIDTH: f32 = 32.0;
+    pub const COLLIDER_HEIGHT: f32 = 29.0;
+
+    pub fn throw(&mut self, force: bool) {
+        self.throwable.throw(&mut self.body, force);
+    }
+
+    pub fn shoot(node: Handle<Galleon>, player: Handle<Player>) -> Coroutine {
+        let coroutine = async move {
+            let player = &mut *scene::get_node(player);
+            scene::add_node(FlyingGalleon::new(player.id));
+
+            player.state_machine.set_state(Player::ST_NORMAL);
+
+            player.weapon = None;
+            scene::get_node(node).delete();
+        };
+
+        start_coroutine(coroutine)
+    }
+
+    pub fn spawn(pos: Vec2) -> HandleUntyped {
+        let mut resources = storage::get_mut::<Resources>();
+
+        let sprite = GunlikeAnimation::new(
+            AnimatedSprite::new(
+                32,
```

Comment:
Constants should be used here. There are a few options, e.g. rename `COLLIDER_*` to something like `SPRITE_*`, or just using `COLLIDER_*` - your choice :slightly_smiling_face: